radix-engine/src/model/method_authorization.rs

```diff
@@ -62,6 +62,16 @@ impl HardResourceOrNonFungible {
     }
 
     pub fn check(&self, auth_zones: &[&AuthZone]) -> bool {
+        // Check if a proof can be virtualized
+        for auth_zone in auth_zones {
+            if let HardResourceOrNonFungible::NonFungible(non_fungible_address) = self {
+                if auth_zone.is_proof_virtualizable(&non_fungible_address.resource_address()) {
```

Comment:
Seems to be wrong abstraction to need to check if proof is virtualized but fine for now. Can we just add a TODO comment that we should move this logic into `AuthZone` at some point